sys/auto_init/can/auto_init_mcp2515.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (C) 2017 OTA keys S.A.
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup sys_auto_init
+ * @{
+ * @file
+ * @brief       initializes mcp2515 can device
+ *
+ * @author      Vincent Dupont <vincent@otakeys.com>
+ * @}
+ */
+
+#ifdef MODULE_MCP2515
+#include "can/device.h"
+#include "mcp2515_params.h"
+
+#define CANDEV_MCP2515_NUMOF ((ARRAY_SIZE(candev_mcp2515_params) / ARRAY_SIZE(candev_params_t)))
+
+#ifndef CANDEV_MCP2515_STACKSIZE
+#define CANDEV_MCP2515_STACKSIZE (THREAD_STACKSIZE_DEFAULT + THREAD_EXTRA_STACKSIZE_PRINTF)
+#endif
+
+#ifndef CANDEV_MCP2515_BASE_PRIORITY
+#define CANDEV_MCP2515_BASE_PRIORITY (THREAD_PRIORITY_MAIN - CANDEV_MCP2515_NUMOF - 2)
```

Comment:
Just wandering what the logic for this priority scheem is?

Reply:
Not sure what the logic behind the priority scheme is (I'm not the author of this part), but this is probably done to allow for multiple CAN devices to be connected to a single controller. 
The candev testapp does not support multiple devices, but conn_can does. 

Anyways, this priority scheme is identical to the one in `sys/auto_init/can/auto_init_periph_can.c` and `sys/auto_init/can/auto_init_can_linux.c` .